test_plans/atomic_ref.asciidoc

```diff
@@ -505,4 +505,264 @@ Check if it returns the original value of the referenced object. Check returned
+* `memory_scope::work_group`
+
+These `ORDER` values which are supported by the device (according to `info::device::atomic_memory_order_capabilities`):
+
+* `memory_order::relaxed`
+* `memory_order::acq_rel`
+* `memory_order::seq_cst`
+
+These `SPACE` values:
+
+* `access::address_space::local_space`
+* `access::address_space::generic_space`
+
+==== Aquire and release
+
+This test uses a `parallel_for` with `nd_range` parameter to atomically update shared scalars in two threads within local group.
+
+[source,c++]
+----
+const TYPE val chaged_val = 42;
+sycl::range group_range(GLOBAL_RANGE/ 2);
+std::array<bool, GLOBAL_RANGE / 2> res;
+vals.fill(false);
+sycl::buffer buf{res.data(), {GLOBAL_RANGE/2}};
+q.submit([&](sycl::handler &cgh) {
+  sycl::accessor res_acc{buf, cgh};
+  sycl::local_accessor<TYPE> x{cgh};
+  sycl::local_accessor<TYPE> y{cgh};
+  cgh.parallel_for_work_group(group_range, sycl::range(2),  [=](auto group) {
+    TYPE A = 0;
+    TYPE B = 0;
+    group.parallel_for_work_item([&](auto item) {
+      sycl::atomic_ref<TYPE, ORDER, SCOPE, SPACE> refA{A};
+      sycl::atomic_ref<TYPE, ORDER, SCOPE, SPACE> refB{B};
+      if (item.get_local_id() == 0) {
+          x[0] = refA.load();
+          refB.store(1);
+        } else {
+          y[0] = refB.load();
+          refA.store(1);
+        }
+     });
+     res_acc[group.get_id()] = !(x[0] == 1 && y[0] == 1);
+    });
+  });
+
+----
+
+Verify that all elements of res is `true` after the loop completes.
+
+This test is repeated for all combinations of supported values listed below:
+
+These `TYPE` types:
+
+* `int`
+* `float`
+* `long long` (if `sizeof(long long) == 8` then only if the device has `aspect::atomic64`)
+* `double` (only if the device has `aspect::fp64` and if `sizeof(double) == 8` only if the device has `aspect::atomic64`)
+
+These `SCOPE` values which are supported by the device (according to `info::device::atomic_memory_scope_capabilities`):
+
+* `memory_scope::device`
+* `memory_scope::system`
+* `memory_scope::work_group`
+
+These `ORDER` values which are supported by the device (according to `info::device::atomic_memory_order_capabilities`):
+
+* `memory_order::acq_rel`
+* `memory_order::seq_cst`
+
+These `SPACE` values:
+
+* `access::address_space::local_space`
+* `access::address_space::generic_space`
+
+==== Ordering
+
+This test uses a `parallel_for` with `nd_range` parameter to atomically read scalar in one thread,
+and then aquire all changes from it in another thread within local group.
+
+[source,c++]
+----
+size_t local_range = q.get_info<sycl::info::device::max_work_group_size>();
+sycl::nd_range nd_range(GLOBAL_RANGE, local_range);
+std::array<bool, GLOBAL_RANGE> res;
+vals.fill(false);
+sycl::buffer buf{res.data(), {GLOBAL_RANGE}};
+q.submit([&](sycl::handler &cgh) {
+  sycl::accessor res_acc{buf, cgh};
+  sycl::local_accessor<TYPE> local_acc{cgh};
+  sycl::local_accessor<bool> arr_acc{{local_range}, cgh};
+  cgh.parallel_for(nd_range, [=](auto item) {
```

Comment:
The values in `arr_acc` are not guaranteed to be initialized to false when the work-group starts.  It might be good to ensure this by adding the following to the start of the kernel:

```
arr_acc[item.get_local_id())] = false;
sycl::group_barrier(item.get_group());
```

Reply:
Done.